src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -357,49 +363,84 @@ impl OpenIDConnectAuthProvider {
-                                            &session.id, err
-                                        );
+                                        match &err {
+                                            // this is where the RFC-6749 5.2 Error Response is received and
+                                            // return to the caller. It's the responsibility of the caller
+                                            // to decide whether to retry or report back to the user.
+                                            //
+                                            // Note that [Errata for RFC 6749](https://www.rfc-editor.org/errata/eid4745)
+                                            // defines two additional error responses, `server_error` and
+                                            // `temporarily_unavailable`, that don't have variant counterparts
+                                            // in the openid-connect crate. These two error messages will
+                                            // therefore **not** end up in the `ServerReponse` variant.
+                                            openidconnect::RequestTokenError::ServerResponse(r) => {
+                                                return Err(r.error().clone());
+                                            }
+                                            openidconnect::RequestTokenError::Request(r) => {
+                                                return Err(CoreErrorResponseType::Extension(format!(
+                                                    "Network Failure while receiving new token: {}",
+                                                    r.to_string()
+                                                )));
+                                            }
+                                            openidconnect::RequestTokenError::Parse(r, _) => {
+                                                return Err(CoreErrorResponseType::Extension(format!(
+                                                    "Error while parsing new token: {}",
+                                                    r.to_string()
+                                                )));
+                                            }
+                                            openidconnect::RequestTokenError::Other(err_string) => {
+                                                match err_string.as_str() {
+                                                    "temporarily_unavailable" | "server_error" => {
+                                                        return Err(CoreErrorResponseType::Extension(
+                                                            err_string.to_string(),
+                                                        ))
+                                                    }
+                                                    _ => {
+                                                        return Err(CoreErrorResponseType::Extension(format!(
```

Comment:
Make `Error` lowercase. Also, remove the #1 as this is not similar to the #2 case below, in this case this is an unexpected response from the 3rd party system or the openid connect Rust crate, while #2 is an internal error in Krill.